services/checks/checks.go

```diff
@@ -139,22 +140,12 @@ func (s *Service) Run(ctx context.Context) {
 	defer ticker.Stop()
 
 	for {
-		var sttEnabled bool
-		settings, err := models.GetSettings(s.db)
-		if err != nil {
-			s.l.Error(err)
-		}
-		if settings != nil && settings.SaaS.STTEnabled {
-			sttEnabled = true
-		}
-
-		if sttEnabled {
-			nCtx, cancel := context.WithTimeout(ctx, checksTimeout)
-			s.collectChecks(nCtx)
-			s.executeChecks(nCtx)
-			cancel()
-		} else {
-			s.l.Info("STT is not enabled, doing nothing.")
+		if err := s.StartChecks(ctx); err != nil {
```

Comment:
Something like that to drop nested `if` and `else`:
```go
err := s.StartChecks(ctx)
switch err {
case nil:
  // nothing, continue
case services.ErrSTTDisabled:
  s.l.Info("STT is not enabled, doing nothing.")
default:
  s.l.Error(err)
}
```

Reply:
Done